autogalaxy/operate/image.py

```diff
@@ -15,53 +17,53 @@ class OperateImage:
     The methods in `OperateImage` are inherited by light objects to provide a concise API.
     """
 
-    def image_2d_from(self, grid: Union[aa.Grid2D, aa.Grid2DIterate]) -> aa.Array2D:
+    def image_2d_from(
+        self,
+        grid: Union[aa.Grid2D, aa.Grid2DIterate],
+        operated_only: Optional[bool] = None,
+    ) -> aa.Array2D:
         raise NotImplementedError
 
-    def blurred_image_2d_via_psf_from(
+    @property
+    def has_light_profile_operated(self) -> bool:
+        raise NotImplementedError
+
+    def _blurred_image_2d_from(
         self,
-        grid: Union[aa.Grid2D, aa.Grid2DIterate],
-        psf: aa.Kernel2D,
-        blurring_grid: Union[aa.Grid2D, aa.Grid2DIterate],
+        image_2d: aa.Array2D,
+        blurring_image_2d: aa.Array2D,
+        psf: Optional[aa.Kernel2D],
+        convolver: aa.Convolver,
     ) -> aa.Array2D:
-        """
-        Evaluate the light object's 2D image from a input 2D grid of coordinates and convolve it with a PSF.
 
-        The input 2D grid may be masked, in which case values outside but near the edge of the mask will convolve light
-        into the mask. A blurring grid is therefore required, which contains image pixels on the mask edge whose light
-        is blurred into the light object's image by the PSF.
+        if psf is not None:
 
-        The grid and blurring_grid must be a `Grid2D` objects so the evaluated image can be mapped to a uniform 2D
-        array and binned up for convolution. They therefore cannot be `Grid2DIrregular` objects.
+            return psf.convolved_array_with_mask_from(
+                array=image_2d.binned.native + blurring_image_2d.binned.native,
+                mask=image_2d.mask,
+            )
 
-        Parameters
-        ----------
-        grid
-            The 2D (y,x) coordinates of the (masked) grid, in its original geometric reference frame.
-        psf
-            The PSF the light object 2D image is convolved with.
-        blurring_grid
-            The 2D (y,x) coordinates neighboring the (masked) grid whose light is blurred into the image.
-        """
+        elif convolver is not None:
 
-        image_2d = self.image_2d_from(grid=grid)
-        blurring_image_2d = self.image_2d_from(grid=blurring_grid)
+            return convolver.convolve_image(
+                image=image_2d, blurring_image=blurring_image_2d
+            )
 
-        return psf.convolved_array_with_mask_from(
-            array=image_2d.binned.native + blurring_image_2d.binned.native,
-            mask=grid.mask,
-        )
+        else:
+
+            raise exc.OperateException(
+                "A PSF or Convolver was not passed to the `blurred_image_2d_list_from()` function."

```

Comment:
Should PSF and Convolver not both be optional with a default of None in that case?